src/java.desktop/share/classes/javax/swing/plaf/synth/SynthComboBoxUI.java

```diff
@@ -352,6 +352,34 @@ public void update(Graphics g, JComponent c) {
         paint(context, g);
     }
 
+    /**
+     * The minimum size is the size of the display area plus insets plus the button.
+     */
+    @Override
+    public Dimension getMinimumSize( JComponent c ) {
+        if ( !isMinimumSizeDirty ) {
+            return new Dimension(cachedMinimumSize);
+        }
+        Dimension size = getDisplaySize();
+        Insets insets = getInsets();
+        Insets arrowInsets = arrowButton.getInsets();
+        //calculate the width and height of the button
+        int buttonHeight = size.height;
+        int buttonWidth = squareButton ?
+                            buttonHeight :
+                            arrowButton.getPreferredSize().width;
+        //adjust the size based on the button width
+        size.height += insets.top + insets.bottom + arrowInsets.top
```

Comment:
Are you sure that the arrow insets should be applied on top of the component's insets, probably the maximum of both should be selected?

Reply:
For combobox text to render correctly arrow inset need to be applied otherwise the combobox editor shows "..." even for a small text. 

Since component inset's return (0, 0, 0, 0) in GTK and Nimbus LAF, didn't take the maximum of both into consideration.

Reply:
@mrserb Are you ok with the fix or do you want anything to add?